src/screen.rs

```diff
@@ -1114,22 +1118,53 @@ impl Screen {
             Cancel => {
                 self.error_file = None;
                 self.set_search(None);
+                self.clear_repeat_count();
                 self.error = None;
                 self.refresh();
                 return DisplayAction::ClearOverlay;
             }
             PreviousFile => return DisplayAction::PreviousFile,
             NextFile => return DisplayAction::NextFile,
-            ScrollUpLines(n) => self.scroll_up(n),
-            ScrollDownLines(n) => self.scroll_down(n),
-            ScrollUpScreenFraction(n) => self.scroll_up_screen_fraction(n),
-            ScrollDownScreenFraction(n) => self.scroll_down_screen_fraction(n),
-            ScrollToTop => self.scroll_to(0),
-            ScrollToBottom => self.following_end = true,
-            ScrollLeftColumns(n) => self.scroll_left(n),
-            ScrollRightColumns(n) => self.scroll_right(n),
-            ScrollLeftScreenFraction(n) => self.scroll_left_screen_fraction(n),
-            ScrollRightScreenFraction(n) => self.scroll_right_screen_fraction(n),
+            ScrollUpLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up(n)
+            }
+            ScrollDownLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_down(n)
+            }
+            ScrollUpScreenFraction(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up_screen_fraction(n)
```

Comment:
For the `_screen_fraction` methods, the value given is the denominator.  This means `10 Shift+Right` will scroll right 1/40th of the screen, rather than 10/4.  You should add a numerator field to all of the `_screen_fraction` methods, and here it will be `self.scroll_up_screen_fraction(self.apply_repeat_count(1), n)`.